src/test/scala/chiselTests/VecToTargetSpec.scala

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests
+
+import chisel3._
+import chisel3.stage.ChiselStage
+
+trait VecToTargetSpecUtils {
+  class Foo extends RawModule {
+    val vec = IO(Input(Vec(4, Bool())))
+
+    // Index a Vec with a Scala literal.
+    val scalaLitIdx = 0
+    val vecScalaLitIdx = vec(scalaLitIdx)
+
+    // Index a Vec with a Chisel literal.
+    val chiselLitIdx = 0.U
+    val vecChiselLitIdx = vec(chiselLitIdx)
+
+    // Index a Vec with a node.
+    val nodeIdx = IO(Input(UInt(2.W)))
+    val vecNodeIdx = vec(nodeIdx)
+
+    // Put an otherwise un-targetable Vec subaccess into a temp.
+    val vecWithTmp = WireInit(vecNodeIdx)
+  }
+
+  var foo: Foo = null
+
+  ChiselStage.elaborate { foo = new Foo; foo }
+
+  val expectedError = "You cannot target Vec subaccess:"
+}
+
```

Comment:
You really don't want to run things like Chisel elaboration during the constructor of a Spec class. It means this code will run before ScalaTest knows what the tests there are to run (Think about each of those `<string> should <string> in { ... }` as test definitions that run before we actually decide which tests to run). Instead, you should use a [Fixture](https://www.scalatest.org/user_guide/sharing_fixtures), or use a testing stile that allows nesting like `FreeSpec` or `FunSpec`, then you can do the elaboration in the outermost block, eg.
```scala
class VecToTargetSpec extends ChiselFunSpec with VecToTargetSpecUtils with Utils {
  describe("VecToTargets") {
    var foo: Foo = null
    ChiselStage.elaborate { foo = new Foo; foo }
    it("Vec subaccess with Scala literal should convert to ReferenceTarget") {
      // Use foo here
    }
  }
}

Reply:
Ok that makes sense. I looked into AnyFunSpec a bit and have reworked this a bit.